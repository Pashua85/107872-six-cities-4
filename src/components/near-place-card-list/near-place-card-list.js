import React from 'react';
import PropTypes from 'prop-types';
import PlaceCardList from '../place-card-list/place-card-list';

const NearPlaceCardList = (props) => {
  return (
    <PlaceCardList className="near-places__list" {...props} />
  );
};

NearPlaceCardList.propTypes = {
  places: PropTypes.arrayOf(PropTypes.object)
};

export default NearPlaceCardList;
