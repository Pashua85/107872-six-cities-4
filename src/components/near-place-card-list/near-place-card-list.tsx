import React from 'react';
import {connect} from 'react-redux';
import {getOffersNearby} from '../../store/reducers/offers-nearby-reducer/selectors';
import PlaceCardList from '../place-card-list/place-card-list';
import {IPlace, IStore} from '../../types/types';

interface NearPlaceCardListProps {
  places: IPlace[]
}

const NearPlaceCardList: React.FC<NearPlaceCardListProps> = (props) => {
  return (
    <PlaceCardList className="near-places__list" {...props} />
  );
};

const mapStateToProps = (state: IStore) => ({
  places: getOffersNearby(state)
});

export default connect(mapStateToProps)(NearPlaceCardList);
export {NearPlaceCardList};
