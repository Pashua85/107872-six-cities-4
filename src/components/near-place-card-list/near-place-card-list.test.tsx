import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {NearPlaceCardList} from './near-place-card-list';
import places from '../../mock/test-places';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`NearPlaceCardList`, () => {
  it(`should render correctly`, () => {
    const wrapper = shallow(
        <NearPlaceCardList places={places} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
