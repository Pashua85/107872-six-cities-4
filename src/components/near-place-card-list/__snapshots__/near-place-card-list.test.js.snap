// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearPlaceCardList should render correctly 1`] = `ShallowWrapper {}`;
